### PR TITLE
Converts world/Topic commands to an extensible decl format.

### PR DESCRIPTION
**Pull request by MarinaGryphon**
- Converts world/Topic commands to an extensible decl format.
  - There is now an option to make localhost world/Topic() requests (from TGS or local scripts) immune to throttling.

This is so that downstreams/modpacks can add their own Topic() commands. For example, the psionics modpack could add a Topic() command to let admins view all active psions without having to edit core game files.

### CODE REVIEW

File: html/changelogs/MDP-declify-topic.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+# Optional: Remove this file after generating master changelog.  Useful for PR changelogs that won't get used again.
+delete-after: True
+
+# Any changes you've made.  See valid prefix list above.
+# INDENT WITH TWO SPACES.  NOT TABS.  SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, all entries are changed into a single [] after a master changelog generation. Just remove the brackets when you add new entries.
+# Please surround your changes in  double quotes ("), as certain characters otherwise screws up compiling. The quotes will not show up in the changelog.
+changes:
```
Comment (Lohikar) [inline html/changelogs/MDP-declify-topic.yml]: No changelogs for backend changes please.

File: config/example/config.txt [hunk 2]
```diff
@@ -450,3 +450,7 @@ RADIATION_LOWER_LIMIT 0.15
 ## clients will still download from the server *too*. This will randomly select one URL if more than one is provided.
 ## Spaces are prohibited in each URL by spec, you must use encoded spaces.
 #RESOURCE_URLS URL URL2 URL3
+
+## Whether or not to make localhost immune to throttling.
+## Localhost will still be throttled internally; it just won't be affected by it.
+#THROTTLE_IMMUNE_LOCALHOST
```
Comment (Lohikar) [inline config/example/config.txt]: `NO_THROTTLE_LOCALHOST` would probably be less awkward.

File: code/game/world_topic_commands.dm [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+	for (var/client/C)
+		x++
+	return x
+
+/decl/topic_command/players
+	name = "players"
+
+/decl/topic_command/players/use()
+	var/n = 0
+	for(var/mob/M in global.player_list)
```
Comment (quardbreak) [inline code/game/world_topic_commands.dm]: ayy lmao legacy code
the entire thing can be replaced with global.clients

**Review by quardbreak — CHANGES_REQUESTED** [review]

File: code/game/world.dm [hunk 4]
```diff
@@ -109,328 +116,15 @@ var/global/world_topic_last = world.timeofday
...
-				var/laws = list()
-				for(var/datum/ai_law/L in lawset_parts[law_type])
-					laws += L.law
-				info[law_type] = list2params(laws)
-
-			info["zero"] = S.laws.zeroth_law ? S.laws.zeroth_law.law : null
-
-			return list2params(info)
+	var/list/params = params2list(T)
+	var/command_key = params[1]
```
Comment (MistakeNot4892) [inline code/game/world.dm:124]: This will runtime if the params are malformed.

File: code/game/world.dm [hunk 5]
```diff
@@ -109,328 +116,15 @@ var/global/world_topic_last = world.timeofday
...
-			for(var/law_type in lawset_parts)
-				var/laws = list()
-				for(var/datum/ai_law/L in lawset_parts[law_type])
-					laws += L.law
-				info[law_type] = list2params(laws)
-
-			info["zero"] = S.laws.zeroth_law ? S.laws.zeroth_law.law : null
-
-			return list2params(info)
+	var/list/params = params2list(T)
```
Comment (MarinaGryphon) [inline code/game/world.dm:121]:
```suggestion
	var/list/params = params2list(T)
	if(!params)
		return
```

File: code/game/world.dm [hunk 6]
```diff
@@ -109,328 +116,17 @@ var/global/world_topic_last = world.timeofday
...
+	if(!params)
```
Comment (MarinaGryphon) [inline code/game/world.dm]:
```suggestion
	if(!length(params))
```
i am too coddled by assoc lists

File: code/game/world_topic_commands.dm [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+/// Initialises the assoc list of topic commands by key.
+/hook/startup/proc/setup_api()
+	for (var/decl/topic_command/TC in decls_repository.get_decls_of_subtype(/decl/topic_command))
+		if(TC.base_type == type)
+			continue
+		global.topic_commands[TC.name] = TC
+	return TRUE
+
+/// Returns a truthy value if we CANNOT use this command.
+/decl/topic_command/proc/can_use(var/T, var/addr, var/master, var/key)
```
Comment (Lohikar) [inline code/game/world_topic_commands.dm:17]: As mentioned on discord, this should probably be renamed.

**Comment by MistakeNot4892** [comment]
This overall looks fine to me, will leave approval to @Lohikar but happy to merge after.

**Review by Lohikar — APPROVED** [review]
Looks fine to me.

**Pull request merged**